octobot_trading/channels/orders.py

```diff
@@ -59,6 +62,10 @@ async def perform(self, orders, is_closed=False, is_from_bot=True):
 
             if symbol is not None:
                 await self._check_missing_open_orders(symbol, orders)
+            if possible_new_order:
+                # possibility new order: refresh portfolio to ensure available funds are up to date
+                await self.channel.exchange_manager.exchange_personal_data.portfolio_manager.\
```

Comment:
There is already a new order detection mechanism. You should put this code in it

Reply:
I did at first. The issue with it is that it will trigger a portfolio refresh for each new order, which is not necessary. Here it will only refresh the portfolio once.

Reply:
I think it should be better to refresh at the end of the open orders refresh mechanism if there is a new order.

Reply:
In Order Updater ? In this case we need somehow to call it on websocket update as well, which doesn't use open order updater. I put it here because it's the common code where a new order will always end up. However maybe it would be better in open orders update right. And then it would be on the websocket to request a balance update if it's not handling it by its websocket updaters. What do you think about it ?